File: backend/ProcesosApi/Models/CampoGlobal.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace ProcesosApi.Models;

public class CampoGlobal
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string Id { get; set; } = null!;
    public string Nombre { get; set; } = string.Empty;
    public string Tipo { get; set; } = "texto"; // "texto", "fecha", "numero"
    public int Orden { get; set; }
    public bool Activo { get; set; } = true;
    public DateTime FechaCreacion { get; set; }
}

File: frontend/src/types/campoGlobal.ts
export interface CampoGlobal {
  id: string;
  nombre: string;
  tipo: string; // "texto" | "fecha" | "numero"
  orden: number;
}

export interface CampoGlobalCreate {
  nombre: string;
  tipo: string;
}

export interface CampoGlobalUpdate {
  nombre: string;
  tipo: string;
}
